api/src/main/scala/org/ensime/api/incoming.scala

```diff
@@ -265,6 +265,24 @@ case class ExpandSelectionReq(file: File, start: Int, end: Int) extends RpcAnaly
  */
 case class StructureViewReq(fileInfo: SourceFileInfo) extends RpcAnalyserRequest
 
+/**
+ * Responds with `ASTInfo`
+ *
+ * @param file source
+ * @param offset in file to inspect
+ */
+case class ASTAtPointReq(
+  file: Either[File, SourceFileInfo],
+  offset: OffsetRange,
+  //A bunch of flags, used to customize output tree format
+  printTypes: Option[Boolean] = None,
```

Comment:
What should sensible flags include? printTypes seems useful, other than that I'm not really sure.